scipy/_lib/_util.py

```diff
@@ -270,6 +270,34 @@ def _asarray_validated(a, check_finite=True,
     return a
 
 
+def _validate_int(k, name, minimum=None):
```

Comment:
I was _just_ thinking about the need for this and wondering why we don't have it already. Look at my last comment in gh-4824:

> Note that I haven't done any input validation on permutations, so negative numbers and incompatible data types raise an error somewhere.

Let's see if we can get this merged so I can use it there!